front-end/virtual_pet/lib/pages/pomodoro_page.dart

```diff
@@ -1,30 +1,124 @@
+  void initState() {
+    super.initState();
+  }
+
+  void _startCountDown() {
+    setState(() {
+      inProgress = true;
+      lastValue = timeLeft;
+    });
+    Timer.periodic(const Duration(seconds: 1), (timer) {
+      if (timeLeft > 0) {
+        setState(() {
+          timeLeft--;
+        });
+      } else {
+        setState(() {
+          inProgress = false;
+        });
+        timer.cancel();
+      }
+    });
+  }
+
+  void _stopTimer() {
+    setState(() {
+      timeLeft = 0;
+      inProgress = false;
+    });
+  }
+
+  void _restartTimer() {
+    setState(() {
+      timeLeft = lastValue;
+    });
+    _startCountDown();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return const Scaffold(
-        /// displays the "Pomodoro Timer" message at the top
-        appBar: CustomAppBar(title: "Pomodoro Timer"),
-        body: Text(
-          "Pomodoro timer coming soon...",
-          style: TextStyle(
-            color: Colors.black,
-            fontSize: 30.0,
+    return Scaffold(
+      // backgroundColor: Colors.grey[900],
+      body: Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          // minutes wheel
+          Container(
+            width: 70,
+            child: ListWheelScrollView.useDelegate(
+              onSelectedItemChanged: (value) {
+                if (inProgress == true) {
+                } else {
+                  setState(() {
+                    timeLeft = value;
+                  });
+                }
+              },
+              itemExtent: 50,
+              perspective: 0.005,
+              diameterRatio: 1.2,
+              physics: const FixedExtentScrollPhysics(),
+              childDelegate: ListWheelChildBuilderDelegate(
+                childCount: 60,
+                builder: (context, index) {
+                  return MyMinutes(
+                    mins: index,
+                  );
+                },
+              ),
+            ),
+          ),
+
+          Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              // Text(inProgress.toString()),
+              // Text("LAST VALUE IS " + lastValue.toString()),
+              Text(
+                timeLeft == 0 ? "DONE" : timeLeft.toString(),
+                style: const TextStyle(fontSize: 50),
+              ),
+              MaterialButton(
+                onPressed: _startCountDown,
+                color: Colors.blue,
+                child: const Text("START TIMER"),
+              ),
+              MaterialButton(
+                onPressed: inProgress == true ? _stopTimer : null,
+                color: Colors.blue,
+                child: const Text("STOP TIMER"),
+              ),
+              MaterialButton(
+                onPressed: inProgress == true ? null : _restartTimer,
+                color: Colors.blue,
+                // child: Text("RESTART TIMER (" + lastValue.toString() + ")"),
```

Comment:
Can delete since you are using string interpolation below to accomplish the same